### ISSUE TITLE
outlook.com images are distorted

### ISSUE THREAD

**Issue by ajankov** [annulen/webkit#776]
Images displayed by new qtwebkit are distorted in https://www.outlook.com . Was working well in the old qtwebkit.

**Comment by annulen** [comment]
Could you provide screenshot?

**Comment by annulen** [comment]
This page uses CSS 3d transforms. Old qtwebkit supports only prefixed CSS properties (`-webkit-transform` etc.), while page uses unprefixed property names, which are supported only by new qtwebkit

**Comment by ajankov** [comment]
Thanks for taking a look at this. I have made the old qtwebkit to also recognize unprefixed 3d transforms and it still does not have that problem I noticed in new qtwebkit.

**Comment by annulen** [comment]
What problem?

**Comment by ajankov** [comment]
The distorted images. There is for example a phone image covering the entire page content which does not happen in the old qtwebkit.

**Comment by annulen** [comment]
Oh, now I understand what you are talking about, it reproduces here

**Comment by annulen** [comment]
It doesn't reproduce when AcceleratedCompositing is enabled, so you can use it as a workaround (in old qtwebkit it was enabled by default)

**Comment by ajankov** [comment]
You are right, but in the old qtwebkit the page is rendered correctly even with AcceleratedCompositing disabled in WebSettings. 

Also, in the new qtwebkit the transformed images quality is very low when AcceleratedCompositing is enabled. Is there any option to improve the quality?

**Comment by ajankov** [comment]
The patch restored the old functionality and the page is rendered well now. Thank you very much for such a quick fix!

**Comment by ajankov** [comment]
Do you think RenderLayer::canRender3DTransforms()  should be updated too in RenderLayer.cpp  ?

In old qtwebkit   :

bool RenderLayer::canRender3DTransforms() const
{
#if USE(ACCELERATED_COMPOSITING)
    return compositor()->canRender3DTransforms();
#elif PLATFORM(QT) && ENABLE(3D_RENDERING)
    return true;
#else
    return false;
#endif
}

In new qtwebkit we have:

bool RenderLayer::canRender3DTransforms() const
{
    return compositor().canRender3DTransforms();
}

**Comment by annulen** [comment]
It can be. Patch above was a quick adaptation of ccee9f651, I didn't even run tests with it yet
